Framework/MDAlgorithms/src/ConvToMDEventsWS.cpp

```diff
@@ -123,20 +125,34 @@ void ConvToMDEventsWS::runConversion(API::Progress *pProgress) {
   // Get the box controller
   Mantid::API::BoxController_sptr bc =
       m_OutWSWrapper->pWorkspace()->getBoxController();
-  size_t lastNumBoxes = bc->getTotalNumMDBoxes();
-  size_t nEventsInWS = m_OutWSWrapper->pWorkspace()->getNPoints();
+
+  // if any property dimension is outside of the data range requested, the job
+  // is done;
+  if (!m_QConverter->calcGenericVariables(m_Coord, m_NDims))
+    return;
+
+  appendEventsFromInputWS(pProgress, bc);
+
+  // m_OutWSWrapper->refreshCentroid();
+  pProgress->report();
+
+  /// Set the special coordinate system flag on the output workspace.
+  m_OutWSWrapper->pWorkspace()->setCoordinateSystem(m_coordinateSystem);
+}
```

Comment:
I've moved `m_OutWSWrapper->pWorkspace()->refreshCache();`  to the `appendEventsFromInputWS` because it is not needed in a new algorithm, caches are constructed during the box structure construction.